modules/auxiliary/scanner/http/f5_mgmt_scanner.rb

```diff
@@ -0,0 +1,107 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'F5 Management Interface Scanner',
+      'Description'   => %q{
+        This module simply detects web management interface of the following F5 Networks devices: BigIP, BigIQ, Enterprise Manager, ARX, and FirePass.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        =>
+        [
+          'Denis Kolegov <dnkolegov[at]gmail.com>',
+          'Oleg Broslavsky <ovbroslavsky[at]gmail.com>',
+          'Nikita Oleksov <neoleksov[at]gmail.com>'
+        ],
+      'DefaultOptions' =>
+        {
+          'SSL' => true,
+          'SSLVersion' => 'TLS1',
+          'RPORT' => 443
+        }
+    ))
+
+    register_options(
+        [
+          OptInt.new('TIMEOUT', [true, "Timeout for the HTTPS probe in milliseconds", 1000])
+        ], self.class)
+  end
+
+  def port_open?(to, verbose)
+    begin
+      ::Timeout.timeout(to) do
```

Comment:
You can use the ```Rex::Proto::Http::Client``` directly to set both the connect and read timeout if it's your concern when scanning a big range of hosts.